cms/src/main/java/uk/nhs/digital/externalstorage/workflow/AbstractExternalFileTask.java

```diff
@@ -77,7 +73,26 @@ private NodeIterator findResourceNodes(Node node) throws RepositoryException {
         return res.getNodes();
     }
 
-    protected abstract void setTargetStatus(S3Connector s3, final String objectKey);
+    protected  NodeIterator findPublicationDatasetsVariant(Node node, String variant) throws RepositoryException {
+        Node folder = node.getParent().getParent();
+
+        String query = format("SELECT * FROM [{0}] WHERE ISDESCENDANTNODE ([''{1}'']) AND [{2}] = ''{3}''",
```

Comment:
The query execution logic is very similar to that in method `findResourceNodes` - there is a good case here to extract a private method for building and executing the query.

It could take the reference node as first argument, the query as the second and a vararg of query arguments (`String... queryArgs`), following the pattern used in this method.

Reply:
extracting ...